lib/rubocop/cop/style/unneeded_interpolation.rb

```diff
@@ -0,0 +1,34 @@
+# encoding: utf-8
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks for strings that are just an interpolated expression.
+      #
+      # @example
+      #
+      #   "#{@var}"
+      class UnneededInterpolation < Cop
+        MSG = 'Prefer `to_s` over string interpolation.'
+
+        INTERPOLATION_TYPES = [
+          :ivar, :cvar, :gvar,
```

Comment:
You can't have an lvar or constant directly embedded in a dstr node without also having a begin node. Otherwise you just get the string for a hashtag. :) `"#lvar"` `"#constant"` These are here to catch the types of interpolation that would violate Style/VariableInterpolation.